### DISCUSSION TITLE
conversion limited to 10 files

### DISCUSSION THREAD

**Discussion by descat85** [ImageMagick/ImageMagick#3907] [Help]
Hello,

I use since a few weeks ImageMagick to convert png files to pdf file with this script:

`#!/bin/bash
echo "file name:"
read file_name
convert *.png $file_name.pdf`

It worked fine until today. Now it is limited to 10 png files, it can't create a pdf file with 12 page for exemple.

My OS is xubuntu 20.04.

Thank you for your help

**Comment by snibgo** [comment]
What version of IM?

>  it can't create a pdf file with 12 page for exemple.

What happens instead? What error messages?

Perhaps you don't have enough memory for all your *.png files. But I don't understand the first part of your bash command, before "convert".

**Reply by descat85** [reply]
version of IM: 8:6.9.10.23+dfsg-2.1ubuntu11.4

The first part of the script ask the pdf file's name and put it in the variable "file_name", but it is not important.

Here is what hapen if i only use `convert *.png file_name.pdf` in a folder wich contains 12 png files
```
****@****:~/Documents/test/trame$ convert *.png file_name.pdf
convert-im6.q16: unable to write pixel cache `/tmp/magick-63156Q1G80jagxfG': Aucun espace disponible sur le périphérique @ error/cache.c/WritePixelCachePixels/5973.
convert-im6.q16: unable to write pixel cache `/tmp/magick-6315k17s51nITv8G': Aucun espace disponible sur le périphérique @ error/cache.c/WritePixelCachePixels/5973.
convert-im6.q16: unable to write pixel cache `/tmp/magick-6315k17s51nITv8G': Aucun espace disponible sur le périphérique @ error/cache.c/WritePixelCachePixels/5973.
convert-im6.q16: unable to write pixel cache `/tmp/magick-6315kfoJusglYyLG': Aucun espace disponible sur le périphérique @ error/cache.c/WritePixelCachePixels/5973.
convert-im6.q16: unable to write pixel cache `/tmp/magick-6315kfoJusglYyLG': Aucun espace disponible sur le périphérique @ error/cache.c/WritePixelCachePixels/5973.
****@****:~/Documents/test/trame$ 
```

"Aucun espace disponible sur le périphérique" = ""No space available on the device" (I am in France)

It creates a pdf file with the 10 first png files

**Comment by snibgo** [comment]
Well, IM says it can't write fles named /tmp/magick-63156Q1G80jagxfG and so on, and it tells you why: you don't have enough free space in /tmp/.

If /tmp/ contains files you don't need, try deleting them.

**Reply by descat85** [reply]
Ok thank you very much! I'm going search this way

**Reply by Jmuccigr** [reply]
You may also want to use a more appropriate tool for turning images into PDFs, like [img2pdf](https://sourceforge.net/projects/img2pdf/).
